### PR TITLE
Implement picture modal when image link is clicked in the comment

### PR DESCRIPTION
**Pull request by imballinst**
This is a WIP PR to implement a picture modal. Prerequisites:

1. https://github.com/imballinst/protosaurus/pull/14
2. https://github.com/imballinst/protosaurus/pull/15
3. https://github.com/imballinst/protosaurus/pull/16

How it works:

1. In the plugin, it parses the syntax `alt` to a button.
2. In the `ProtosaurusImage` component, we add event listeners to all buttons with a matching class name (and yes, we remove these events on unmount).
3. On button click (which is made look like a text, later we'll add an icon), it will open a modal.
4. The modal shows the image and the close button. We can also click on the backdrop to close the modal. Later, we'll add escape button to also close the modal.

https://user-images.githubusercontent.com/7077157/153540721-81905b16-7165-4b21-a93c-ef13b38ae1b6.mov

## To-dos

- [x] Add icon to the button so that it indicates a "picture"
- [x] Add "Escape" key event listener to the modal
- [x] Move the z-index more above the navbar

Signed-off-by: Try Ajitiono <ballinst@gmail.com>

### CODE REVIEW

**Comment by imballinst** [comment]
The conflicts I will address after the other PR has been merged.

**Comment by dio** [comment]
I have merged all of the pre-reqs, feel free to continue this when you have time. Thanks, @imballinst!

**Comment by imballinst** [comment]
Done. Thanks for merging the other PRs @dio! Let me know if there are feedback that you might have.

File: mdx/lib/mdx.ts [hunk 1]
```diff
@@ -15,14 +15,16 @@ export async function emitMdx(filePath: string, pkg: PackageData) {
...
 
-import ReferenceWrapper from "@theme/ReferenceWrapper";
 import Description from "@theme/Description";
 import Definition from "@theme/Definition";
 import DefinitionHeader from "@theme/DefinitionHeader";
 import RpcDefinition from "@theme/RpcDefinition";
 import RpcDefinitionHeader from "@theme/RpcDefinitionHeader";
 import RpcDefinitionDescription from "@theme/RpcDefinitionDescription";
 import RpcMethodText from "@theme/RpcMethodText";
+import ProtosaurusImage from "@theme/ProtosaurusImage";
```
Comment (dio) [inline mdx/lib/mdx.ts:25]: Sorry if this question comes late: so, do we already have a plan on how users will have this "extension"? I mean, when they init a docusaurus project what's the next step? (if done manually).

Seems like through plugin mechanism, so the plugin can come as a separate npm module?

**Review by dio — APPROVED** [review]
Awesome! Just one future direction question.

File: mdx/lib/mdx.ts (hunk 1 above)
Comment (imballinst) [reply mdx/lib/mdx.ts:25]: Ah, yes, that's a very good point. I haven't thought of that at the moment. But yeah, I think theme plugin is the way to do it. For example: https://github.com/facebook/docusaurus/tree/main/packages/docusaurus-theme-classic

Then _maybe_, in users' `docusaurus.config.js`, they do this:

```
module.exports = {
  ...
+ themes: ['@protosaurus/theme-protosaurus'],
  ...
}
```

The same can be done for the `proto-messages` plugin, which users will add to the `rehypePlugins`.

In that case, perhaps we can have a `packages` folder, containing `theme-protosaurus` and `rehype-protosaurus`, where both plugins will be published to npm? We can think about using stuff such as Lerna after that.

Comment (dio) [reply mdx/lib/mdx.ts:25]: SGTM. Thank you!

**Pull request merged**